### ISSUE TITLE
[Feature Request] Z-Probe Offset

### ISSUE THREAD

**Issue by KenLucke** [cp2004/OctoPrint-EEPROM-Marlin#3]
When you get into the tasks you have set for yourself, please add the Z-Probe Offset into the parameters that are read/set (this is different, of course, from the simple Z-Offset already included).

And thank you for taking this over!

**Comment by cp2004** [comment]
I can look into this, but may need testing when its implemented before release since I don't have a printer with a probe.

**Comment by invisiblek** [comment]
The regex is wrong for newer versions of marlin (2.0.0 and beyond i think, but i need to dig through commit history) so the options just don't show up. They now include probe X/Y offsets too. The whole version detection stuff in this plugin is a shit show though. I've hacked it to work on the old repo (since i was only ever running it on my marlin build), but will need to be retooled to be compatible with all versions. I can look in to this if I get some time.

**Comment by cp2004** [comment]
> The whole version detection stuff in this plugin is a shit show though

I can agree with you there (why compare *individual version strings*??!), there are a few things like this that make me scratch my head. 

The last update the plugin saw before I took over maintenance last week, was 2018 (before 2.x was really *a thing*) so it is on my list in #1 to test with the latest versions of Marlin (I have 2.0.5.3 available, and looking at putting EEPROM into the Virtual Printer in OctoPrint so I could simulate this)

Thanks for looking into it!

**Comment by cp2004** [comment]
Apologies that this has taken such a long time to release - I am in progress of re-writing this plugin from the ground up, and it has taken longer than I initially thought to get around to, than when I decided not to release a new version before then. Its on its way, the data collection and sending is done, it now needs a UI. Hopefully it will make it before the end of 2020 still. Stay tuned!

**Comment by cp2004** [comment]
@KenLucke @invisiblek

Please test the release candidate of V3 of this plugin, it is available and this feature/fix is implemented.

Details can be found in the feedback ticket of how to install, update and report issues: #16

**Comment by KenLucke** [comment]
I have tried SEVERAL times to use the URL https://github.com/cp2004/OctoPrint-EEPROM-Marlin/archive/3.0.0r3.zip to try to install via the Plugin Manager.  All it does is sit there and look at me.  It never installs, never sends notices, refuses to enable or accept the close button... nothing.  I have to reload OP to get out of the install screen.

![CR-10 v2 (Yellow) LAN  OctoPrint  2020-12-14 14-32-50](https://user-images.githubusercontent.com/57936294/102143841-671d2580-3e19-11eb-8755-669b7f3128a1.jpg)


I've restarted OP, rebooted the whole system, still unable to install the RC.
